Dockerfile

```diff
@@ -1,20 +1,28 @@
 # syntax=docker/dockerfile:1
 
-FROM python:3.7-slim-buster as builder
+ARG  PY_VERSION=3.10
+FROM python:${PY_VERSION}-buster as builder
 
-WORKDIR /app
+WORKDIR /py4ai-core
+
+RUN apt-get update && apt-get upgrade -y
 
 COPY requirements/requirements.in requirements/requirements.in
-COPY requirements/requirements_dev.txt requirements/requirements_dev.txt
-COPY LICENSE MANIFEST.in versioneer.py setup.py pyproject.toml README.md ./
+COPY requirements/requirements_dev.in requirements/requirements_dev.in
+COPY LICENSE MANIFEST.in versioneer.py setup.py pyproject.toml README.md Makefile ./
 COPY py4ai py4ai
 COPY tests tests
-RUN pip install -r requirements/requirements_dev.txt
-RUN python3 -m pytest
-RUN python3 setup.py sdist
+
+RUN addgroup --system tester && adduser --system --group tester
+RUN chown -R tester:tester /py4ai-core
+
+# change to the tester user: switch to a non-root user is a best practice.
+USER tester
+ENV PATH ${PATH}:/home/tester/.local/bin
+RUN make checks
 
 FROM python:3.7-slim-buster
-WORKDIR /app
-COPY --from=builder /app/dist /app/dist
+WORKDIR /py4ai-core
+COPY --from=builder /py4ai-core/dist /py4ai-core/dist
 RUN ls -t ./dist/*.tar.gz | xargs pip install
 ENTRYPOINT ["python"]
```

Comment:
Can we introduce a USER here (and fix related permissions)? I believe it is way more important than the user in the builder phase. 

The best-practice of using non-root is because of security when running the image, not when building (although why not having a USER there too, but it is not so crucial)

Reply:
Done